common/models/model-method.js

```diff
@@ -10,4 +11,58 @@ module.exports = function(ModelMethod) {
    * @class ModelMethod
    * @inherits WorkspaceEntity
    */
+
+  ModelMethod._shouldEncodeStaticFlagInName = function() {
+    var version = ModelMethod.app.models.Workspace.loopBackVersion;
+    return version != null ? !semver.gtr('3.0.0', version) : false;
+  };
+
+  ModelMethod.getJsonKey = function(data) {
+    if (!this._shouldEncodeStaticFlagInName()) {
+      return data.name;
+    }
+    var isStatic = data.isStatic;
+    if (isStatic !== undefined) {
+      var name = data.name;
+      var matchName = name.match(/^prototype\.(.*)$/);
+      if (!isStatic && (matchName === null || !matchName)) {
+        data.name = 'prototype.' + name;
+      }
+    }
+
+    return data.name;
+  };
+
+  ModelMethod.getConfigFromData = function(data) {
+    var properties = this.definition.properties;
+    var result = {};
+    var prop;
+
+    // add pre-defined properties in the order defined by LDL
+    // apply `json` config from LDL along the way
+    for (prop in properties) {
+      if (properties[prop].json === false) continue;
+      result[properties[prop].json || prop] = data[prop];
+    }
```

Comment:
Can we please not copy this code and call reuse `ModelMethod.base.getConfigFromData` instead?